ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,172 @@
+        )
+        try:
+            return json.loads(response.content)
+        except JSONDecodeError as json_exception:
+            LOGGER.error(
+                "Exception converting query response to JSON: "
+                f"exc={json_exception} response_status_code={response.status_code} response={response.content}"
+            )
+            raise
+
+    def query(self, query):
+        return self._get_response(query=f"{self.api_v1}/query?query={query}")
+
+    def get_alert(self, alert_name):
+        """
+        For getting alert metrics
+
+        Args:
+             alert(str): alert name
+             Ex. result = prometheus.get_alert(alert='WatchDog')
+
+        Returns:
+             dict containing alert metrics for each matching alert
+        """
+        response = self.alerts
+        matching_alerts = {}
+        alert_list = []
+        for alert in response["data"]["alerts"]:
+
+            if alert["labels"]["alertname"] == alert_name:
+                alert_list.append(alert)
+        matching_alerts[alert_name] = alert_list
+        return matching_alerts
+
+    def alert_sampler(self, alert, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for alert function and raises timeout exception
+
+        Args:
+             alert(str): alert name
+             timeout: wait time, default is 10 mins
+
+        Raise:
+             TimeoutExpiredError: if alert
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=1,
+            func=self.get_alert,
+            alert=alert,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                if sample and sample[0]["value"][-1] == "1":
+                    return
+        except TimeoutExpiredError:
+            LOGGER.error(
+                f"Failed to get successful alert {alert}. Current data: {sample}"
+            )
+            raise
+
+    def query_sampler(self, query, timeout=TIMEOUT_10MIN):
+        """
+        Sample output for query function
+
+        Args:
+             query(str):
+             wait_timeout(int): default is 10 mins
+        Return:
+             return the query result as a list
+
+        Raise:
+             TimeoutExpiredError: if query response doesn't return success
+        """
+        sampler = TimeoutSampler(
+            wait_timeout=timeout,
+            sleep=1,
+            func=self.query,
+            query=query,
+        )
+        sample = None
+        try:
+            for sample in sampler:
+                result = sample["data"]["result"]
+                if sample["status"] == "success":
+                    return result
+        except TimeoutExpiredError:
+            LOGGER.error(
+                f"Failed to get successful status after executing query '{query}'. Current data: {sample}"
+            )
+            raise
+
+    def verify_prometheus_connection(self):
```

Comment:
Where this is used?

Reply:
removed it for now, i'll add it after confirmation